packages/fluentui/react-builder/src/components/Designer.tsx

```diff
@@ -342,6 +344,43 @@ export const Designer: React.FunctionComponent = () => {
 
   const [{ mode, isExpanding, isSelecting }, setMode] = useMode();
   const [showJSONTree, handleShowJSONTreeChange] = React.useState(false);
+  const [axeErrors, setAxeErrors] = React.useState([]);
+
+  React.useEffect(() => {
+    if (state.selectedJSONTreeElementUuid) {
```

Comment:
you have quite a lot of logic under this guard, would it make sense to invert the condition and return early ?

Reply:
I would even leave the if here and move the rest to a separate function.